engine/access/rest/scripts.go

```diff
@@ -0,0 +1,81 @@
+package rest
+
+import (
+	"fmt"
+
+	"github.com/onflow/flow-go/access"
+	"github.com/onflow/flow-go/engine/access/rest/generated"
+)
+
+// executeScript handler sends the script from the request to be executed.
+func executeScript(r *request, backend access.API, _ LinkGenerator) (interface{}, error) {
+	blockID := r.getQueryParam("block_id")
+	blockHeight := r.getQueryParam("block_height")
+
+	// validate block ID and block height is provided
+	if blockID == "" && blockHeight == "" {
+		return nil, NewBadRequestError(fmt.Errorf("either block ID or block height must be provided"))
+	}
+	if blockID != "" && blockHeight != "" {
+		return nil, NewBadRequestError(fmt.Errorf("can not provide both block ID and block height"))
+	}
+	// parse body to script
+	var scriptBody generated.ScriptsBody
+	err := r.bodyAs(&scriptBody)
+	if err != nil {
+		return nil, err
+	}
+
+	args, err := toScriptArgs(scriptBody)
+	if err != nil {
+		return nil, NewBadRequestError(err)
+	}
+
+	code, err := toScriptSource(scriptBody)
+	if err != nil {
+		return nil, NewBadRequestError(err)
+	}
+
+	if blockID != "" {
+		id, err := toID(blockID)
+		if err != nil {
+			return nil, NewBadRequestError(err)
+		}
+
+		result, err := backend.ExecuteScriptAtBlockID(r.Context(), id, code, args)
+		if err != nil {
+			return nil, err
+		}
+
+		return result, nil
```

Comment:
Yes it could, I remember being corrected to not do that from someone but I don't know what the reasoning was and I can't find anything by googling so I will just revert to simplified one liner.

Reply:
one reason that I know of is that you don't want to return a half baked result object if there is an error so you make sure you intentionally return a nil for the result if err !=nil.